ComponentKit/Components/CKTestTypedPropsComponent.h

```diff
@@ -0,0 +1,24 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <ComponentKit/CKTypedPropsComponent.h>
+
+struct CKTestTypedPropsComponentProps {
+  CKRequiredProp<NSString *> string;
+  UIFont *font;
+};
+
+@interface CKTestTypedPropsComponent : CKTypedPropsComponent
+
++ (instancetype)newWithProps:(const CKTestTypedPropsComponentProps &)props
+                        view:(const CKComponentViewConfiguration &)view
+                        size:(const CKComponentSize &)size;
+
+@end
```

Comment:
This is just here as a demonstration. In order to provide type-safety, you have to provide your own +new that does a pass-through to the parent's +new method.